components/core/src/os/users/admincheck.c

```diff
@@ -0,0 +1,90 @@
+#include <windows.h>
+#include <winnt.h>
+#include <ShlObj.h>
+
+#pragma comment (lib,"shell32.lib")
+
+TOKEN_ELEVATION_TYPE s_elevationType = TokenElevationTypeDefault;
+BOOL                 s_bIsAdmin = FALSE;
+
+// returns 0 if an admin
+// returns 1 if not
+int GetElevationState()
+{
+    if (GetProcessElevation(&s_elevationType, &s_bIsAdmin)) {
+        switch(s_elevationType) {
+            // Default user or UAC is disabled
+            case TokenElevationTypeDefault:  
+                if (IsUserAnAdmin()) {
+                    return 0;
```

Comment:
Honestly not trying to bike shed here, but I think returning early is a great thing.  Keeping track of the state of your function all the way through, just for the purpose of having a single return at the end feels like additional complexity for no real benefit.